cmd/controller/main.go

```diff
@@ -50,6 +50,63 @@ var (
 	workerCount   int
 )
 
+// TODO:desc
+type controllerContext struct {
+	// TODO: desc
+	kubeClient kubernetes.Interface
+
+	// TODO: desc
+	extClient apiextclient.Interface
+
+	// TODO: desc
+	machineClient machineclientset.Interface
+	// TODO add leaderElectionClient
+
+	// TODO: desc
+	sshKeyPair *ssh.PrivateKey
+
+	// TODO: desc
+	ips []net.IP
+
+	// TODO: desc
+	metrics *MachineControllerMetrics
+
+	// TODO: desc
+	kubeInformerFactory kubeinformers.SharedInformerFactory
+
+	// TODO: desc
+	machineInformerFactory machineinformers.SharedInformerFactory
+
+	// TODO: desc
+	stopCh <-chan struct{}
+}
+
+// TODO: desc
+func createClientsOrDie(cfg *rest.Config) (kubernetes.Interface, apiextclient.Interface, machineclientset.Interface) {
```

Comment:
I think it's fine to create these clients simply in the main func. We don't need to wrap these things.
Plus: Do we really want to return interfaces? You should always return concrete types. Other funcs can then ask for Interfaces.